components/common/src/templating/config.rs

```diff
@@ -1150,7 +1159,13 @@ mod test {
         let contents = "foo\nbar\n";
 
         assert_eq!(file.exists(), false);
-        write_templated_file(&file, &contents, &USER, &GROUP).expect("should fail on permissions");
+        match write_templated_file(&file, &contents, &curr_username(), &curr_groupname()) {
+            Ok(_) => panic!("should fail on permissions"),
+            Err(e) => match e {
+                Error::PermissionFailed(e) => assert_eq!(file.to_string_lossy(), e.to_string()),
+                wrong => panic!("Unexpected error returned: {:?}", wrong),
+            },
+        };
```

Comment:
Unless there are places in the code where we specifically handle `PermissionFailed` differently, I think this test (and the code above to generate the `PermissionFailed` error) is overly-specific. I'm not sure why we should be concerned about the details beyond the fact that `write_templated_file` should fail. That leads to a test which is less likely to break in the future if we change some specifics that don't invalidate the overall behavior. Plus, it makes the code simpler:
```rust
        assert!(
            write_templated_file(&file, &contents, &curr_username(), &curr_groupname()).is_err()
        );
```

Reply:
Yeah, I was trying to get better error messages during the test failure.  the `io::Error` we get back doesn't give much info, but I think it's too much for this one specific case. Happy to roll back the error mapping code